subsystem/kubeapi_test.go

```diff
@@ -564,6 +565,86 @@ func randomNameSuffix() string {
 	return fmt.Sprintf("-%s", strings.Split(uuid.New().String(), "-")[0])
 }
 
+func printCRs(ctx context.Context, client k8sclient.Client) {
+	if GinkgoT().Failed() {
+		var (
+			multiErr              *multierror.Error
+			aciList               hiveext.AgentClusterInstallList
+			agentList             v1beta1.AgentList
+			infraEnvList          v1beta1.InfraEnvList
+			bareMetalHostList     bmhv1alpha1.BareMetalHostList
+			nmStateConfigList     v1beta1.NMStateConfigList
+			clusterImageSetList   hivev1.ClusterImageSetList
+			clusterDeploymentList hivev1.ClusterDeploymentList
+		)
+
+		multiErr = multierror.Append(multiErr, client.List(ctx, &agentList, k8sclient.InNamespace(Options.Namespace)))
+		aList, err := json.Marshal(agentList)
+		if err != nil {
+			multiErr = multierror.Append(multiErr, err)
+		} else {
+			_, err = GinkgoWriter.Write([]byte(fmt.Sprintf("failed test '%s' created the agents: %s\n", GinkgoT().Name(), aList)))
```

Comment:
Wrap it in a function that will get a string or format
I think that it can be treated as a log function

Reply:
Done